### PR TITLE
[T3A3][W13-A4] Ngin Yun Chuan

### PR DESCRIPTION
**Pull request by nginyc**
Ready for review

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -45,8 +46,8 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
 > Persons can have any number of tags (including 0)
 
 Examples: 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
-* `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
+* `add John Doe p/98765432 e/johnd@gmail.com a/123, Clementi Ave 3, #12-34, 231534`
+* `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/123, Clementi Ave 3, #12-34, 231534 t/criminal t/friend`
```
Comment (damithc) [inline doc/UserGuide.md]: Good job updating user docs

File: src/seedu/addressbook/commands/AddCommand.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
             + "Contact details can be marked private by prepending 'p' to the prefix.\n"
             + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
             + "Example: " + COMMAND_WORD
-            + " John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney";
+            + " John Doe p/98765432 e/johnd@gmail.com a/" + Address.EXAMPLE + " t/friends t/owesMoney";
```
Comment (damithc) [inline src/seedu/addressbook/commands/AddCommand.java]: Nice.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -7,12 +7,29 @@
...
+    public static final String EXAMPLE = 
+    		Block.EXAMPLE + ADDRESS_PARTS_DELIMITER + " " 
+    		+ Street.EXAMPLE + ADDRESS_PARTS_DELIMITER + " "
+    		+ Unit.EXAMPLE + ADDRESS_PARTS_DELIMITER + " " 
+    		+ PostalCode.EXAMPLE;
+    		
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "A person's address must be in format: BLOCK, STREET, UNIT, POSTAL_CODE";
+    
+    // address format is 4 strings delimited by strictly 3 commas
+    public static final String ADDRESS_VALIDATION_REGEX = "([.]|[^,])*,([.]|[^,])*,([.]|[^,])*,([.]|[^,])*";
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Address.java]: Good to see trying to use regex.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -25,31 +42,67 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        // Split address string to parts
+        String[] addressParts = address.split(ADDRESS_PARTS_DELIMITER);
+        _block = new Block(addressParts[ADDRESS_PARTS_INDEX_BLOCK].trim());
+        _street = new Street(addressParts[ADDRESS_PARTS_INDEX_STREET].trim());
+        _unit = new Unit(addressParts[ADDRESS_PARTS_INDEX_UNIT].trim());
+        _postalCode = new PostalCode(addressParts[ADDRESS_PARTS_INDEX_POSTALCODE].trim());
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Address.java]: While our coding standard allows this naming format (i.e. prefix local variables with underscore), you should be consistent with rest of the code which uses a different naming convention.

**Comment by damithc** [comment]
Good job @nginyc 
Some comments added. Please close the PR after reading comments.

**Comment by nginyc** [comment]
Got it, thanks!

**Pull request closed without merging**